generator/src/main/java/org/corfudb/generator/operations/CheckpointOperation.java

```diff
@@ -28,10 +29,10 @@ public void execute() {
 
             MultiCheckpointWriter mcw = new MultiCheckpointWriter();
             mcw.addAllMaps(state.getMaps());
-            long trimAddress = mcw.appendCheckpoints(state.getRuntime(), "Maithem");
-            state.setTrimMark(trimAddress);
+            Token trimAddress = mcw.appendCheckpoints(state.getRuntime(), "Maithem");
+            state.setTrimMark(trimAddress.getSequence());
             Thread.sleep(1000l * 30l * 1l);
-            state.getRuntime().getAddressSpaceView().prefixTrim(trimAddress - 1);
+            state.getRuntime().getAddressSpaceView().prefixTrim(trimAddress.getSequence() - 1);
```

Comment:
I replaced setTrimMark with an updateTrimMark method, but for prefixTrim(), gc(), invalidateClientCache and invalidateServerCaches I left them as is because they don't really belong to the State.

Reply:
ok